lib/modules/manager/npm/extract/index.ts

```diff
@@ -255,8 +265,8 @@ export async function extractAllPackageFiles(
           });
         }
       } else {
-        if (packageFile.endsWith('json')) {
-          logger.trace({ packageFile }, `Extracting as a package.json file`);
+        if (packageFile.startsWith('package.')) {
```

Comment:
As you have described users could add a JSON5 file in here and then your changes to parsing and  replacement logic will no longer work. 

```suggestion
        if (regEx('package\\.(json?|ya?ml)$').test(packageFile)) {
```

Can we move these kind of logic into the pnpm directory. 
This looks messy.

Reply:
ok, you mean that it will crash trying to parse the json5 file, right? I will try to test something, maybe we can give support to json5 also.